PMM-12153 Update mysqld_exporter to upstream

ademidoff opened the pull request.

MattiasAng committed:
Support for scraping multiple mysqld hosts (#651)

Add multi-target exporter scrape support.

Breaking change:
`DATA_SOURCE_NAME` is now removed. Local configuration is now supplied by `MYSQLD_EXPORTER_PASSWORD` and command line arguments.

Signed-off-by: Mattias Ängehov <mattias.angehov@castoredc.com>

bgarret committed:
Fix README formatting

Add a missing code block closing tag

Signed-off-by: Benoit Garret <benoit.garret_github@gadz.org>

greut committed:
fix: log message without any key (#564)

> {"(go=go1.16.4, user=root@e2043849cb1f, date=20210531-07:30:16)":"(MISSING)","caller":"mysqld_exporter.go:278","level":"info","msg":"Build context","ts":"2021-07-27T07:51:05.601Z"}

This is the log we see and it doesn't please Kibana.

Signed-off-by: Yoan Blanc <yblanc@edgelab.ch>

Signed-off-by: Yoan Blanc <yblanc@edgelab.ch>
Co-authored-by: Ben Kochie <superq@gmail.com>

prombot committed:
Update common Prometheus files (#666)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

Signed-off-by: prombot <prometheus-team@googlegroups.com>

DaAwesomeP committed:
Update to exporter-toolkit v0.8.1 (#677)

* bump exporter-toolkit to v0.8.1

Signed-off-by: Perry Naseck <git@perrynaseck.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_golang from 1.13.0 to 1.14.0 (#681)

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.13.0 to 1.14.0.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.13.0...v1.14.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump gopkg.in/ini.v1 from 1.66.6 to 1.67.0 (#670)

Bumps [gopkg.in/ini.v1](https://github.com/go-ini/ini) from 1.66.6 to 1.67.0.
- [Release notes](https://github.com/go-ini/ini/releases)
- [Commits](https://github.com/go-ini/ini/compare/v1.66.6...v1.67.0)

---
updated-dependencies:
- dependency-name: gopkg.in/ini.v1
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

cezmunsta committed:
Fix infoSchemaInnodbMetricsEnabledColumnQuery

The query introduced with #523 performs poorly when there
are sufficient numbers of columns in `information_schema.columns`.

* Added using the `table_schema` as a filter
* Added limit to the query

Signed-off-by: cezmunsta <github@incoming-email.co.uk>

SuperQ committed:
Update Go (#688)

* Update Go

* Update Go to 1.19.
* Update some Go modules.

Signed-off-by: SuperQ <superq@gmail.com>

dependabot[bot] committed:
Bump github.com/prometheus/exporter-toolkit from 0.8.1 to 0.8.2 (#690)

Bumps [github.com/prometheus/exporter-toolkit](https://github.com/prometheus/exporter-toolkit) from 0.8.1 to 0.8.2.
- [Release notes](https://github.com/prometheus/exporter-toolkit/releases)
- [Changelog](https://github.com/prometheus/exporter-toolkit/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prometheus/exporter-toolkit/compare/v0.8.1...v0.8.2)

---
updated-dependencies:
- dependency-name: github.com/prometheus/exporter-toolkit
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

ericyan committed:
Fix unclosed code block in README.md (#691)

This fixes a code block that not properly closed, causing the next two sections rendered as code.

Signed-off-by: Eric Yan <github@ericyan.me>

Signed-off-by: Eric Yan <github@ericyan.me>

vokey committed:
docs: fix prometheus scrape example in README

Fix the type of params in Prometheus scrape_config

Signed-off-by: ☮ Vokey <lexypig@gmail.com>

yosida95 committed:
add support for unix:// target to the multi-target scraping

Signed-off-by: Kohei YOSHIDA <kohei@yosida95.com>

soara committed:
using mysql.Config.FormatDSN() in FormDSN() (#708)

Signed-off-by: Yasushi MIYAZAKI <MIYAZAKI.Yasushi@gmail.com>

LeoQuote committed:
fix duplicated collector when collectParams was passed in probe mode (#710)

* fix duplicated collector when collectParams was passed in probe mode

---------

Signed-off-by: qizhicheng <qizhicheng@douban.com>

dependabot[bot] committed:
Bump github.com/go-sql-driver/mysql from 1.6.0 to 1.7.0 (#702)

Bumps [github.com/go-sql-driver/mysql](https://github.com/go-sql-driver/mysql) from 1.6.0 to 1.7.0.
- [Release notes](https://github.com/go-sql-driver/mysql/releases)
- [Changelog](https://github.com/go-sql-driver/mysql/blob/master/CHANGELOG.md)
- [Commits](https://github.com/go-sql-driver/mysql/compare/v1.6.0...v1.7.0)

---
updated-dependencies:
- dependency-name: github.com/go-sql-driver/mysql
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

prombot committed:
Update common Prometheus files (#695)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

SuperQ committed:
Update build (#711)

* Update Go to 1.20
* Update golangci-lint.
* Use LTS releases for mariadb testing.
* Update CI orb.

Signed-off-by: SuperQ <superq@gmail.com>

prombot committed:
Update common Prometheus files (#712)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

yosida95 committed:
Fix a mixture of error messages and metrics is returned from the
multiple target endpoint.

If the config file doesn't contain a section requested by the
`auth_module` parameter and/or a network address requested by the
`target` parameter is ill-formed, emit one line of error message only.

This changes not-yet-released behaviour that emits both error messages
and metrics.

This commit also removes `probe_success` metrics, also not yet released,
for the following reasons:

  - Existing `/metrics` endpoint does not contain it.
  - It will always be 1 even if failed to connect to the
    MySQL server.
  - Existing `mysql_up` parameter can be used instead.

Signed-off-by: Kohei YOSHIDA <kohei@yosida95.com>

SuperQ committed:
Update exporter-toolkit

Update to the latest exporter-toolkit.
* Switch to new kingpin path.
* Use new landing page feature.
* Fixup version handling in tests.

Signed-off-by: SuperQ <superq@gmail.com>

SuperQ committed:
Merge pull request #715 from prometheus/superq/update_toolkit

Update exporter-toolkit

dependabot[bot] committed:
Bump github.com/prometheus/common from 0.41.0 to 0.42.0

Bumps [github.com/prometheus/common](https://github.com/prometheus/common) from 0.41.0 to 0.42.0.
- [Release notes](https://github.com/prometheus/common/releases)
- [Commits](https://github.com/prometheus/common/compare/v0.41.0...v0.42.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/common
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

SuperQ committed:
Merge pull request #716 from prometheus/dependabot/go_modules/github.com/prometheus/common-0.42.0

Bump github.com/prometheus/common from 0.41.0 to 0.42.0

SuperQ committed:
Merge pull request #705 from vokey/main

Fix prometheus scrape example in README.md

SuperQ committed:
Merge pull request #714 from yosida95/early-return

Fix a mixture of error messages and metrics is returned from the multiple target endpoint.

SuperQ committed:
Merge pull request #659 from bgarret/patch-2

Fix README formatting

SuperQ committed:
Release v0.15.0-rc.0

BREAKING CHANGES:

The exporter no longer supports the monolithic `DATA_SOURCE_NAME` environment variable.
To configure connections to MySQL you can either use a `my.cnf` style config file or command line arguments.

For example:

    export MYSQLD_EXPORTER_PASSWORD=secret
    mysqld_exporter --mysqld.address=localhost:3306 --mysqld.username=exporter

We have also dropped some internal scrape metrics:
* `mysql_exporter_scrapes_total`
* `mysql_exporter_scrape_errors_total`
* `mysql_last_scrape_failed`

Changes:

* [CHANGE] Allow tlsCfg.InsecureSkipVerify outside of mTLS #631
* [CHANGE] Update to exporter-toolkit v0.8.1 #677
* [CHANGE] Fix shared metrics between requests #722
* [FEATURE] Add support for collecting metrics from sys.user_summary #628
* [FEATURE] Support for multi-target mysqld probes #651
* [FEATURE] Add MySQL TLS configurations #718
* [BUGFIX] Fix infoSchemaInnodbMetricsEnabledColumnQuery #687

Signed-off-by: SuperQ <superq@gmail.com>

soara committed:
Add MySQL TLS configurations (#718)

This PR is a modified version of the #674 to match the FormDSN supported by the #708.

Fixes: https://github.com/prometheus/mysqld_exporter/issues/673

Signed-off-by: Nico Braun <rainbowstack@gmail.com>
Signed-off-by: Yasushi MIYAZAKI <MIYAZAKI.Yasushi@gmail.com>
Co-authored-by: Nico Braun <rainbowstack@gmail.com>

prombot committed:
Update common Prometheus files (#720)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

ejortegau committed:
Add support for collecting metrics from sys.user_summary (#628)

* Add support for collecting metrics from sys.user_summary

Signed-off-by: Eduardo J. Ortega U <5791035+ejortegau@users.noreply.github.com>
Signed-off-by: Eduardo J. Ortega U. <5791035+ejortegau@users.noreply.github.com>
Co-authored-by: Ben Kochie <superq@gmail.com>

LeoQuote committed:
Fix shared metrics between requests (#722)

## Breaking changes

### dropped metrics
* mysql_exporter_scrapes_total 
* mysql_exporter_scrape_errors_total
* mysql_last_scrape_failed 

### new metrics
* mysql_exporter_collector_success (replace mysql_exporter_scrape_errors_total)

### metrics isolated between requests
* mysql_up
* mysql_exporter_collector_success
* mysql_exporter_collector_duration_seconds

## Why is this neccessary
We are using mysqld_exporter in `multi-target` method, which means one exporter for multiple mysql instance, the crucial metric `mysql_up` must be isolated between requests, or there will be misunderstandings, some mysql instance would shown as down while in fact it is not.

all shared metrics are dropped especially `mysql_exporter_scrapes_total`  and `mysql_exporter_scrape_errors_total` , the metrics are useless since prometheus can record almost everything, including scrape fail for each collector.

`mysql_last_scrape_failed`  is covered by `mysql_exporter_collector_success` so it is also dropped.

fixes #713 

---------

Signed-off-by: qizhicheng <qizhicheng@douban.com>

SuperQ committed:
Refactor mysql_up metric

Use a return value on the scrape function to only send the `mysql_up`
metric down the channel once.

Signed-off-by: SuperQ <superq@gmail.com>

SuperQ committed:
Merge pull request #723 from prometheus/superq/scrape_up

Refactor mysql_up metric

SuperQ committed:
Merge pull request #687 from cezmunsta/fix-infoSchemaInnodbMetricsEnabledColumnQuery

Fix infoSchemaInnodbMetricsEnabledColumnQuery

SuperQ committed:
Merge pull request #707 from yosida95/unix-domain-socket

Add UNIX domain socket support for multi-target scraping

SuperQ committed:
Merge pull request #717 from prometheus/superq/0.15.0-rc.0

Release v0.15.0-rc.0

SuperQ committed:
Fix CircleCI tag building

Match any `v...` tags for releases.

Signed-off-by: SuperQ <superq@gmail.com>

tinpotnick committed:
Use target in DSN builder.
Signed-off-by: Nick Knight <nick@babblevoice.com>

ColinDKelley committed:
issue-#725: use STRAIGHT_JOIN in infoSchemaAutoIncrementQuery

Signed-off-by: Colin Kelley <colindkelley@gmail.com>

Commit:
Add config.my-cnf hot reload `/-/reload`

Signed-off-by: wangzhen30 <wangzhen30@lenovo.com>

towo committed:
Correct `mysql_info_schema_threads` in dashboard

The dashboard example still uses `mysql_info_schema_threads`, which had
been changed to `mysql_infa_schema_processlist_total` in
4bbd404549555bf5e5059f3b322094fb62d155c2 and then renamed to
`mysql_infa_schema_processlist_threads` in 17460a409ec0f62fbb58664db6059f667b88e2a6.

Signed-off-by: Tobias Wolter <tobias.wolter@b1-systems.de>

Commit:
feat: logTargetWhileError

Signed-off-by: 黄振明 <huangzhenming@qunjielong.com>

ats-frosmo committed:
use float64 for latencies, uint64 is not large enough

Signed-off-by: Aki Salakka <aki.salakka@frosmo.com>

SuperQ committed:
Rename subsystem const

Avoid conflict with `mysql` package name.

Signed-off-by: SuperQ <superq@gmail.com>

SuperQ committed:
Merge pull request #738 from prometheus/superq/fix_subsystem

Rename subsystem const

SuperQ committed:
Update Prometheus Go modules

* Update Prometheus Go modules to latest.
* Fixup landing page in test.

Signed-off-by: SuperQ <superq@gmail.com>

Commit:
feat: rename xmysql to mysql

Signed-off-by: 黄振明 <huangzhenming@qunjielong.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_golang from 1.14.0 to 1.15.1

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.14.0 to 1.15.1.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.14.0...v1.15.1)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

SuperQ committed:
Merge pull request #736 from mingregister/logtarget

feat: log Target  while error

SuperQ committed:
Merge pull request #741 from prometheus/dependabot/go_modules/github.com/prometheus/client_golang-1.15.1

Bump github.com/prometheus/client_golang from 1.14.0 to 1.15.1

dependabot[bot] committed:
Bump github.com/prometheus/client_model from 0.3.0 to 0.4.0

Bumps [github.com/prometheus/client_model](https://github.com/prometheus/client_model) from 0.3.0 to 0.4.0.
- [Release notes](https://github.com/prometheus/client_model/releases)
- [Commits](https://github.com/prometheus/client_model/compare/v0.3.0...v0.4.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_model
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

SuperQ committed:
Merge pull request #740 from prometheus/dependabot/go_modules/github.com/prometheus/client_model-0.4.0

Bump github.com/prometheus/client_model from 0.3.0 to 0.4.0

SuperQ committed:
Merge pull request #739 from prometheus/superq/toolkit

Update Prometheus Go modules

SuperQ committed:
Merge pull request #737 from singularity-xx/main

BUGFIX for 0.15.0-rc.0  sys.X$user_summary latencies doesn't fit to uint64

winfredwz committed:
Merge branch 'prometheus:main' into main

winfredwz committed:
check ReloadConfig error

Signed-off-by: winfredwz <winfredwz@163.com>

winfredwz committed:
check ReloadConfig error

Signed-off-by: winfredwz <winfredwz@163.com>

SuperQ committed:
Merge pull request #728 from babblevoice/bug-target-ignored

Use target in DSN builder.

dependabot[bot] committed:
Bump github.com/smartystreets/goconvey from 1.7.2 to 1.8.0

Bumps [github.com/smartystreets/goconvey](https://github.com/smartystreets/goconvey) from 1.7.2 to 1.8.0.
- [Release notes](https://github.com/smartystreets/goconvey/releases)
- [Commits](https://github.com/smartystreets/goconvey/compare/v1.7.2...v1.8.0)

---
updated-dependencies:
- dependency-name: github.com/smartystreets/goconvey
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump github.com/go-sql-driver/mysql from 1.7.0 to 1.7.1

Bumps [github.com/go-sql-driver/mysql](https://github.com/go-sql-driver/mysql) from 1.7.0 to 1.7.1.
- [Release notes](https://github.com/go-sql-driver/mysql/releases)
- [Changelog](https://github.com/go-sql-driver/mysql/blob/master/CHANGELOG.md)
- [Commits](https://github.com/go-sql-driver/mysql/compare/v1.7.0...v1.7.1)

---
updated-dependencies:
- dependency-name: github.com/go-sql-driver/mysql
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

SuperQ committed:
Merge pull request #735 from b1-systems/fix/grafana-example

Correct `mysql_info_schema_threads` in dashboard

SuperQ committed:
Merge pull request #726 from Invoca/prometheus-issue-#725-info_schema_auto_increment-with-straight_join

issue-#725: use STRAIGHT_JOIN in infoSchemaAutoIncrementQuery

SuperQ committed:
Merge pull request #732 from prometheus/dependabot/go_modules/github.com/smartystreets/goconvey-1.8.0

Bump github.com/smartystreets/goconvey from 1.7.2 to 1.8.0

SuperQ committed:
Update mysqld_exporter.go

Fixup nits

Signed-off-by: Ben Kochie <superq@gmail.com>

SuperQ committed:
Merge pull request #730 from prometheus/dependabot/go_modules/github.com/go-sql-driver/mysql-1.7.1

Bump github.com/go-sql-driver/mysql from 1.7.0 to 1.7.1

SuperQ committed:
Allow empty passwords

Empty passwords are allowed for UNIX socket connections and when using
TLS auth.

Fixes: https://github.com/prometheus/mysqld_exporter/issues/686

Signed-off-by: SuperQ <superq@gmail.com>

SuperQ committed:
Merge pull request #734 from winfredwz/main

Add config.my-cnf hot reload `/-/reload`

prombot committed:
Update common Prometheus files

Signed-off-by: prombot <prometheus-team@googlegroups.com>

SuperQ committed:
Merge pull request #743 from prometheus/repo_sync

Synchronize common files from prometheus/prometheus

SuperQ committed:
Merge pull request #742 from prometheus/superq/empty_password

Allow empty passwords

SuperQ committed:
Don't use HOME env in the my-cnf config (#745)

* Don't use HOME env in the my-cnf config

Use a relative file as the default for the .my.cnf style configuration.

Fixes: https://github.com/prometheus/mysqld_exporter/issues/633

---------

Signed-off-by: SuperQ <superq@gmail.com>
Signed-off-by: Ben Kochie <superq@gmail.com>
Co-authored-by: Matthias Rampke <matthias.rampke@googlemail.com>

SuperQ committed:
Release 0.15.0 (#744)

BREAKING CHANGES:

The exporter no longer supports the monolithic `DATA_SOURCE_NAME` environment variable.
To configure connections to MySQL you can either use a `my.cnf` style config file or command line arguments.

For example:

    export MYSQLD_EXPORTER_PASSWORD=secret
    mysqld_exporter --mysqld.address=localhost:3306 --mysqld.username=exporter

We have also dropped some internal scrape metrics:
- `mysql_exporter_scrapes_total`
- `mysql_exporter_scrape_errors_total`
- `mysql_last_scrape_failed`

The default client configuration file is now `.my.cnf` in the process working directory. Use `--config.my-cnf="$HOME/.my.cnf"` to retain the previous default.

Changes:

* [CHANGE] Allow `tlsCfg.InsecureSkipVerify` outside of mTLS #631
* [CHANGE] Update to exporter-toolkit v0.8.1 #677
* [CHANGE] Fix shared metrics between requests #722
* [CHANGE] Allow empty passwords #742
* [CHANGE] Don't use HOME env in the my-cnf config path. #745
* [FEATURE] Add support for collecting metrics from `sys.user_summary` #628
* [FEATURE] Support for multi-target mysqld probes #651
* [FEATURE] Add MySQL TLS configurations #718
* [FEATURE] Add config reload via /-/reload #734
* [ENHANCEMENT] Add UNIX domain socket support for multi-target scraping #707
* [ENHANCEMENT] Use `STRAIGHT_JOIN` in infoSchemaAutoIncrementQuery #726
* [BUGFIX] Fix `infoSchemaInnodbMetricsEnabledColumnQuery` #687
* [BUGFIX] Allow empty passwords #742

Signed-off-by: SuperQ <superq@gmail.com>

dependabot[bot] committed:
Bump github.com/smartystreets/goconvey from 1.8.0 to 1.8.1 (#751)

Bumps [github.com/smartystreets/goconvey](https://github.com/smartystreets/goconvey) from 1.8.0 to 1.8.1.
- [Release notes](https://github.com/smartystreets/goconvey/releases)
- [Commits](https://github.com/smartystreets/goconvey/compare/v1.8.0...v1.8.1)

---
updated-dependencies:
- dependency-name: github.com/smartystreets/goconvey
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_golang from 1.15.1 to 1.16.0 (#752)

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.15.1 to 1.16.0.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.15.1...v1.16.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

prombot committed:
Update common Prometheus files (#748)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

prombot committed:
Update common Prometheus files (#753)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

prombot committed:
Update common Prometheus files (#762)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_golang from 1.16.0 to 1.17.0 (#772)

Signed-off-by: Ben Kochie <superq@gmail.com>

dependabot[bot] committed:
Bump github.com/google/uuid from 1.3.0 to 1.3.1 (#768)

Bumps [github.com/google/uuid](https://github.com/google/uuid) from 1.3.0 to 1.3.1.
- [Release notes](https://github.com/google/uuid/releases)
- [Changelog](https://github.com/google/uuid/blob/master/CHANGELOG.md)
- [Commits](https://github.com/google/uuid/compare/v1.3.0...v1.3.1)

---
updated-dependencies:
- dependency-name: github.com/google/uuid
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

prombot committed:
Update common Prometheus files (#767)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

prombot committed:
Update common Prometheus files (#773)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

SuperQ committed:
Switch spell check to golangci-lint (#792)

Switch from using codespell to misspell.

Signed-off-by: SuperQ <superq@gmail.com>

dependabot[bot] committed:
Bump github.com/alecthomas/kingpin/v2 from 2.3.2 to 2.4.0 (#789)

Bumps [github.com/alecthomas/kingpin/v2](https://github.com/alecthomas/kingpin) from 2.3.2 to 2.4.0.
- [Release notes](https://github.com/alecthomas/kingpin/releases)
- [Commits](https://github.com/alecthomas/kingpin/compare/v2.3.2...v2.4.0)

---
updated-dependencies:
- dependency-name: github.com/alecthomas/kingpin/v2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/common from 0.44.0 to 0.45.0 (#779)

Bumps [github.com/prometheus/common](https://github.com/prometheus/common) from 0.44.0 to 0.45.0.
- [Release notes](https://github.com/prometheus/common/releases)
- [Commits](https://github.com/prometheus/common/compare/v0.44.0...v0.45.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/common
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/google/uuid from 1.3.1 to 1.4.0 (#782)

Bumps [github.com/google/uuid](https://github.com/google/uuid) from 1.3.1 to 1.4.0.
- [Release notes](https://github.com/google/uuid/releases)
- [Changelog](https://github.com/google/uuid/blob/master/CHANGELOG.md)
- [Commits](https://github.com/google/uuid/compare/v1.3.1...v1.4.0)

---
updated-dependencies:
- dependency-name: github.com/google/uuid
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

prombot committed:
Update common Prometheus files (#786)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

dependabot[bot] committed:
Bump github.com/google/go-cmp from 0.5.9 to 0.6.0 (#780)

Bumps [github.com/google/go-cmp](https://github.com/google/go-cmp) from 0.5.9 to 0.6.0.
- [Release notes](https://github.com/google/go-cmp/releases)
- [Commits](https://github.com/google/go-cmp/compare/v0.5.9...v0.6.0)

---
updated-dependencies:
- dependency-name: github.com/google/go-cmp
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

SuperQ committed:
Fixup mixin linting (#793)

Fix linter errors for the mixin dashboard.

Signed-off-by: SuperQ <superq@gmail.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_model (#781)

Bumps [github.com/prometheus/client_model](https://github.com/prometheus/client_model) from 0.4.1-0.20230718164431-9a2bf3000d16 to 0.5.0.
- [Release notes](https://github.com/prometheus/client_model/releases)
- [Commits](https://github.com/prometheus/client_model/commits/v0.5.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_model
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

blkperl committed:
Update golang to 1.21 (#791)

Signed-off-by: william.vanhevelingen <william.vanhevelingen@acquia.com>
Co-authored-by: Ben Kochie <superq@gmail.com>

blkperl committed:
Release v0.15.1 (#794)

Signed-off-by: william.vanhevelingen <william.vanhevelingen@acquia.com>

prombot committed:
Update common Prometheus files (#795)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

it-percona-cla commented:
[![CLA assistant check](https://cla-assistant.percona.com:443/pull/badge/not_signed)](https://cla-assistant.percona.com:443/percona/mysqld_exporter?pullRequest=181) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.percona.com:443/percona/mysqld_exporter?pullRequest=181) before we can accept your contribution.<br/>**0** out of **6** committers have signed the CLA.<br/><br/>:x: SuperQ<br/>:x: prombot<br/>:x: mmontes11<br/>:x: s10<br/>:x: mtanda<br/>:x: OscarVanL<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com:443/check/percona/mysqld_exporter?pullRequest=181) it.</sub>

dependabot[bot] committed:
Bump github.com/prometheus/common from 0.45.0 to 0.46.0 (#809)

Bumps [github.com/prometheus/common](https://github.com/prometheus/common) from 0.45.0 to 0.46.0.
- [Release notes](https://github.com/prometheus/common/releases)
- [Commits](https://github.com/prometheus/common/compare/v0.45.0...v0.46.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/common
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/DATA-DOG/go-sqlmock from 1.5.0 to 1.5.2 (#810)

Bumps [github.com/DATA-DOG/go-sqlmock](https://github.com/DATA-DOG/go-sqlmock) from 1.5.0 to 1.5.2.
- [Release notes](https://github.com/DATA-DOG/go-sqlmock/releases)
- [Commits](https://github.com/DATA-DOG/go-sqlmock/compare/v1.5.0...v1.5.2)

---
updated-dependencies:
- dependency-name: github.com/DATA-DOG/go-sqlmock
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

prombot committed:
Update common Prometheus files (#796)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

dependabot[bot] committed:
Bump github.com/google/uuid from 1.4.0 to 1.6.0 (#808)

Bumps [github.com/google/uuid](https://github.com/google/uuid) from 1.4.0 to 1.6.0.
- [Release notes](https://github.com/google/uuid/releases)
- [Changelog](https://github.com/google/uuid/blob/master/CHANGELOG.md)
- [Commits](https://github.com/google/uuid/compare/v1.4.0...v1.6.0)

---
updated-dependencies:
- dependency-name: github.com/google/uuid
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/exporter-toolkit from 0.10.0 to 0.11.0 (#801)

Bumps [github.com/prometheus/exporter-toolkit](https://github.com/prometheus/exporter-toolkit) from 0.10.0 to 0.11.0.
- [Release notes](https://github.com/prometheus/exporter-toolkit/releases)
- [Changelog](https://github.com/prometheus/exporter-toolkit/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prometheus/exporter-toolkit/compare/v0.10.0...v0.11.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/exporter-toolkit
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

prombot committed:
Update common Prometheus files (#811)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

prombot committed:
Update common Prometheus files (#817)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

OscarVanL committed:
Update docker registry link in README.md (#813)

Signed-off-by: Oscar van Leusen <oscarvanleusen@gmail.com>

prombot committed:
Update common Prometheus files (#820)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_golang from 1.18.0 to 1.19.0 (#823)

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.18.0 to 1.19.0.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/v1.19.0/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.18.0...v1.19.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_model from 0.5.0 to 0.6.0 (#822)

Bumps [github.com/prometheus/client_model](https://github.com/prometheus/client_model) from 0.5.0 to 0.6.0.
- [Release notes](https://github.com/prometheus/client_model/releases)
- [Commits](https://github.com/prometheus/client_model/compare/v0.5.0...v0.6.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_model
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

prombot committed:
Update common Prometheus files (#825)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

mmontes11 committed:
Support for prometheus scrape timeout in probe endpoint (#828)

Signed-off-by: Martin Montes <martin11lrx@gmail.com>

prombot committed:
Update common Prometheus files (#829)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

dependabot[bot] committed:
Bump github.com/go-sql-driver/mysql from 1.7.1 to 1.8.1 (#830)

Bumps [github.com/go-sql-driver/mysql](https://github.com/go-sql-driver/mysql) from 1.7.1 to 1.8.1.
- [Release notes](https://github.com/go-sql-driver/mysql/releases)
- [Changelog](https://github.com/go-sql-driver/mysql/blob/v1.8.1/CHANGELOG.md)
- [Commits](https://github.com/go-sql-driver/mysql/compare/v1.7.1...v1.8.1)

---
updated-dependencies:
- dependency-name: github.com/go-sql-driver/mysql
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

prombot committed:
Update common Prometheus files (#832)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

mtanda committed:
Support MySQL 8.4 replicas syntax (#837)

* support MySQL 8.4

Signed-off-by: Mitsuhiro Tanda <mitsuhiro.tanda@gmail.com>
---------

Signed-off-by: Mitsuhiro Tanda <mitsuhiro.tanda@gmail.com>

prombot committed:
Update common Prometheus files (#834)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

SuperQ committed:
Update build (#838)

* Update Go to 1.22.
* Update Go modules.
* Update version collector.
* Update integration tests.

Signed-off-by: SuperQ <superq@gmail.com>

prombot committed:
Update common Prometheus files (#839)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

s10 committed:
Add the instance struct to handle connections (#859)

The intent is to use the instance struct to hold the connection
to the database as well as metadata about the instance:
- version
- flavor (mariadb or mysql)

Change is similar to prometheus-community/postgres_exporter#785

Signed-off-by: Vlad Gusev <vlad.esten@gmail.com>

dependabot[bot] committed:
Bump github.com/prometheus/common from 0.53.0 to 0.55.0 (#851)

Bumps [github.com/prometheus/common](https://github.com/prometheus/common) from 0.53.0 to 0.55.0.
- [Release notes](https://github.com/prometheus/common/releases)
- [Changelog](https://github.com/prometheus/common/blob/main/RELEASE.md)
- [Commits](https://github.com/prometheus/common/compare/v0.53.0...v0.55.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/common
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_golang from 1.19.0 to 1.19.1 (#845)

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.19.0 to 1.19.1.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.19.0...v1.19.1)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
